packages/remix-oxygen/src/server.ts

```diff
@@ -1,21 +1,37 @@
 import {
+  ActionArgs,
   createRequestHandler as createRemixRequestHandler,
+  LoaderArgs,
   type AppLoadContext,
   type ServerBuild,
 } from '@remix-run/server-runtime';
+import type {Logger} from '@shopify/hydrogen';
 
 export function createRequestHandler<Context = unknown>({
   build,
   mode,
   poweredByHeader = true,
   getLoadContext,
+  log,
 }: {
   build: ServerBuild;
   mode?: string;
   poweredByHeader?: boolean;
   getLoadContext?: (request: Request) => Promise<Context> | Context;
+  log?: Logger;
 }) {
-  const handleRequest = createRemixRequestHandler(build, mode);
+  const routes: typeof build.routes = {};
+
+  for (const [id, route] of Object.entries(build.routes)) {
```

Comment:
This is where remix modules loader and actions are overwritten

Reply:
I'm not sure if it's the job of remix-oxygen to wrap the Remix build and enhance it with logs.
Maybe we could also consider exporting this as a standalone function from Hydrogen itself?
Perhaps as part of `createLogger` (as an option, you pass the Remix build to enhance its logs), or as a separate utility?